src/AFQMC/Hamiltonians/HSPotential_Helpers.h

```diff
@@ -23,15 +23,17 @@ namespace afqmc
 {
 namespace HamHelper
 {
+template<class T>
 std::vector<std::size_t> count_nnz_per_cholvec(double cut, TaskGroup_& TG, SpVType_shm_csr_matrix& V2, int NMO);
 
+template<class T>
```

Comment:
in `config.h` replace 
```
using SpCType_shm_csr_matrix =
    ma::sparse::csr_matrix<SPComplexType, int, std::size_t, shared_allocator<SPComplexType>, ma::sparse::is_root>;
using SpVType_shm_csr_matrix =
    ma::sparse::csr_matrix<SPValueType, int, std::size_t, shared_allocator<SPValueType>, ma::sparse::is_root>;
using CType_shm_csr_matrix =
    ma::sparse::csr_matrix<ComplexType, int, std::size_t, shared_allocator<ComplexType>, ma::sparse::is_root>;
using VType_shm_csr_matrix =
    ma::sparse::csr_matrix<ValueType, int, std::size_t, shared_allocator<ValueType>, ma::sparse::is_root>;
```
with
```
template<typename T>
using CSRMtrxShared<T> = ma::sparse::csr_matrix<SPComplexType, int, std::size_t, shared_allocator<SPComplexType>, ma::sparse::is_root>;
```
In most cases this will just read
`CSRMatrixShared<Value>` or `CSRMatrixShared<Complex>` which is much easier to read quickly and surfaces the important parameter for both read and specializing code.

then one can write
```
template<typename T>
...(...
     CSRMatrix<T>  WorthlessTwoLetterVariableName,
     ...)
{
....
if constexpr (IsComplex_t<T>::value) {
  ...
}
...
}
```